web/src/components/tasks/ElevatorRegistrationStatusSummary.tsx

```diff
@@ -23,49 +25,86 @@ export const ElevatorRegistrationStatusSummary = (props: Props): ReactElement =>
 
   const { business } = useUserData();
 
-  const getInformationalMessageText = (): string => {
-    const statuses: string[] = [];
-    for (const reg of props.summary.registrations) {
-      statuses.push(reg.status);
+  type CardDetails = {
+    headerColor: string
+    bodyColor: string;
+    icon: string;
+    iconTextColor: string;
+  }
+
+  const getDetailsForRegistrationCard = (status: string): CardDetails => {
```

Comment:
I know multiple switches over the same bits of information is a bit of an anti-pattern, but I couldn't find a better way of doing it.

The first switch statement covers general formatting. The second switch statement handles messages only relevant to certain statuses. The third switch statement only covers a unique case with a single status and is related to a special formatting case. 

If anyone has ideas on a better way to organize this, I'm all ears

Reply:
I considered that (the `CardDetails` object was sort of starting this) but I think that would be harder to read/maintain. In that case, we would have to lay out every possible status instead of being able to batch them and make good use of the `default` case. 

But I'm not opposed to it if that's what the consensus is!

Reply:
> I considered that (the `CardDetails` object was sort of starting this) but I think that would be harder to read/maintain. In that case, we would have to lay out every possible status instead of being able to batch them and make good use of the `default` case.
> 
> But I'm not opposed to it if that's what the consensus is!

For sure, which is why I understand going the `switch` route. I think I would have done the same thing. I was just thinking about your comment and thought of a map, because it seems to be the approach when someone doesn't want to use a switch. 

I'm excited about this convo! 😄